spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/ConverterAwareMappingSpannerEntityWriter.java

```diff
@@ -393,6 +405,12 @@ private void writeProperty(MultipleValueBinder sink,
 				valueSet = attemptSetSingleItemValue(Value.COMMIT_TIMESTAMP, Timestamp.class, valueBinder,
 						Timestamp.class, this.writeConverter);
 			}
+			// annotated json column, bind directly
+			else if (property.getAnnotatedColumnItemType() != null &&
+					property.getAnnotatedColumnItemType().equals(Type.Code.JSON)) {
+				valueBinder.to(jsonToValueConverter(propertyValue));
```

Comment:
I think it may be preferable to keep all the stuff about the ".to() method" in one of the `attemptSetSingleValue` or `attemptSetSingleItemValue` method.

It would make the code much more parallel I think since it's the responsibility of one of those methods to invoke the correct `.to()` method on the binder. You can input the check in there on the property type if it's JSON to select the correct binder.

Reply:
Good point. I am adding `annotatedColumnItemType` to the parameter of `attemptSetSingleItemValue` since this one is private, so that the Json logic goes inside.